rust/src/io/object_store.rs

```diff
@@ -122,12 +123,16 @@ impl ObjectStore {
         &self.base_path
     }
 
+    /// Open a file for path
     pub async fn open(&self, path: &Path) -> Result<Box<dyn ObjectReader>> {
-        Ok(Box::new(CloudObjectReader::new(
-            self,
-            path.clone(),
-            self.prefetch_size,
-        )?))
+        match self.scheme.as_str() {
+            "file" => LocalObjectReader::open(path),
```

Comment:
i forgot, if the path does not contain explicit scheme://, it's automatically interpreted as file:// ?

Reply:
`scheme` is set in the ObjectStore level (only once)

https://github.com/eto-ai/lance/blob/main/rust/src/io/object_store.rs#L63-L66

So either no `scheme` or `scheme == file://` are treated as LocalFileSystem